lib/ci-helper.ts

```diff
@@ -626,15 +638,43 @@ export class CIHelper {
         }
     }
 
+    public async checkCommits(pr: IPullRequestInfo):
+        Promise<string> {
+        let comment: string = "";
+        const maxCommits = 30;
+        if (pr.commits && pr.commits > maxCommits) {
+            comment = `The pull request has ${pr.commits
+                } commits.  The max allowed is ${maxCommits
+                }.  Please squash some commits and 'git push --force'`;
```

Comment:
I would rather suggest to split the patch series.

Squashing commits would probably result in multiple unrelated changes within the same commits (which is something that the reviewers will point out as a problem later on).

Reply:
I am open to any suggested verbiage.

_Please split the patch series into multiple pull requests.  Also consider squashing related commits._

Key here is to provide direction and use wording that is clear to everyone.  Would it be better to have a wiki page to discuss limiting the number of commits?  The message could then just point to the wiki page (the wiki page could be as detailed as needed or I am just punting responsibility).  Hmm, thinking a submission guidelines page may be useful so the linting errors can also point to it.  What do you think?  Don't want the rejection messages to be too verbose so pointing to (soon to be) existing doc may be a good way to go.

Reply:
I know there is doc on git patch submission but a wiki page would be more specific on the gitgitgadget guidelines/linting.

Reply:
> Please split the patch series into multiple pull requests. Also consider squashing related commits.

I like that.

The wiki page idea is also good, maybe have both the message _and_ a link to the wiki page?

Reply:
There is a wiki page on git-for-windows/git regarding commits.  Should it be expanded regarding linting or should the gitgitgadget wiki have this info?

Reply:
Quite honestly, I'd rather move the wiki page from git-for-windows to gitgitgadget, leaving a link to the latter in place of the former.